web/src/unread_ops.ts

```diff
@@ -622,6 +622,8 @@ export function mark_topic_as_read(stream_id: number, topic: string): void {
         ],
         "add",
     );
+    message_lists.current?.resume_reading();
+    unread_ui.hide_unread_banner();
```

Comment:
Hmm, I think this needs to be in the `success` callback, after checking we're still looking at the view. There's two problems:
- You could be calling this from the sidebar menu when looking at another view.
- The network requests could fail.

Reply:
Pushed a commit for that in #33300